aardvark-app/src/window.rs

```diff
@@ -237,13 +247,37 @@ mod imp {
                 self,
                 move |_| {
                     let buffer = this.open_document_entry.buffer();
-                    let input_len = buffer
+                    let input: String = buffer
                         .text(&buffer.start_iter(), &buffer.end_iter(), false)
                         .chars()
                         .filter(|c| c.is_digit(16))
-                        .count();
+                        .collect();
+
+                    this.open_document_button.set_sensitive(input.len() == 64);
 
-                    this.open_document_button.set_sensitive(input_len == 64);
+                    let existing = if input.len() == 64 {
+                        let app = this
+                            .obj()
+                            .application()
+                            .and_then(|app| app.downcast::<AardvarkApplication>().ok())
+                            .expect("Application needs to be a AardvarkApplication");
+                        app.window_for_document_id(&input)
+                    } else {
+                        None
+                    };
+
+                    if let Some(window) = existing {
+                        if window == *this.obj() {
+                            this.open_document_button
+                                .set_label(&gettext("Document already open"));
```

Comment:
@bertob the text probably isn't great for the case when the document is already opened in the current window.

Reply:
![image](https://github.com/user-attachments/assets/015e0c17-282f-42c2-bd81-28ec0ad851d8)

![image](https://github.com/user-attachments/assets/9b6f2ae1-3f39-43c3-8e86-e914b60b78a0)

Hmmm, how about "Switch to Existing Window" and just disabling the button if it's the same window? At least for now, in a later step we could maybe add an additional label below the entry giving a bit more context.

Btw, button labels should always use Title Case ;)